### PR TITLE
refactor(core): NULL -> nullptr

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Null -> nullptr

## Tests Performed:
- Builds

## How to Test the Changes:

CI

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Review by FrancescoBorzi — APPROVED** [review]

File: src/common/Packets/ByteBuffer.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ ByteBufferSourceException::ByteBufferSourceException(size_t pos, size_t size,
     std::ostringstream ss;
 
     ss << "Attempted to put a "
-       << (valueSize > 0 ? "NULL-pointer" : "zero-sized value")
+       << (valueSize > 0 ? "nullptr-pointer" : "zero-sized value")
```
Comment (Kitzunu) [inline src/common/Packets/ByteBuffer.cpp:32]: Could this cause issues if it is nullptr-pointer?

@Winfidonarleyan @Helias

File: src/server/game/Tools/PlayerDump.cpp [hunk 2]
```diff
@@ -392,11 +392,11 @@ DumpReturn PlayerDumpWriter::WriteDump(const std::string& file, uint32 guid)
 
 void fixNULLfields(std::string& line)
 {
-    std::string nullString("'NULL'");
+    std::string nullString("'nullptr'");
     size_t pos = line.find(nullString);
     while (pos != std::string::npos)
     {
-        line.replace(pos, nullString.length(), "NULL");
+        line.replace(pos, nullString.length(), "nullptr");
```
Comment (Kitzunu) [inline src/server/game/Tools/PlayerDump.cpp:399]: here too

**Comment by Winfidonarleyan** [comment]
Plz check file 'src/common/Debugging/WheatyExceptionReport.cpp' in TrinityCore

File: src/common/Packets/ByteBuffer.cpp (hunk 1 above)
Comment (Winfidonarleyan) [reply src/common/Packets/ByteBuffer.cpp:32]: I think no

Comment (Viste) [reply src/common/Packets/ByteBuffer.cpp:32]: бля жесть чел, Null-pointer = nullptr и получается ты пишешь nullpointer-pointer :D

File: src/server/game/Tools/PlayerDump.cpp (hunk 2 above)
Comment (Viste) [reply src/server/game/Tools/PlayerDump.cpp:399]: нет никакого смысла в nullptr тут, можно просто оставить NULL. Угомони свою автозамену :)

Comment (Kitzunu) [reply src/server/game/Tools/PlayerDump.cpp:399]: You know I don't speak russian Viste D: :P

Comment (Viste) [reply src/server/game/Tools/PlayerDump.cpp:399]: > You know I don't speak russian Viste D: :P

sorry I was too lazy to write in English :D

Comment (Viste) [reply src/server/game/Tools/PlayerDump.cpp:399]: if in brief this replacement makes no sense at all

File: src/common/Packets/ByteBuffer.cpp (hunk 1 above)
Comment (Viste) [reply src/common/Packets/ByteBuffer.cpp:32]: if in brief this replacement makes no sense at all

File: src/server/game/Tools/PlayerDump.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/game/Tools/PlayerDump.cpp:399]: My thought too, will revert those two lines
Thanks 💚

**Comment by Kitzunu** [comment]
Done

**Review by Viste — APPROVED** [review]

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**